# Issue 20180: Bug: is_locally_represented_number_at_place returns incorrect value

archive/issues_019943.json:
```json
{
    "body": "Sorry this bug report is a bit ad hoc, but I suspect the thing that makes this buggy is part of a larger problem in the quadratic forms module.  I'm running Sage 6.9 and I get the following in error if I construct the quadratic form \n\n```\nM=QuadraticForm(ZZ,3,[75,120,120,144,0,144])\n```\nthen \n\n```\nsage: M.is_locally_represented_number_at_place(99,3)\nFalse\n```\nand \n\n```\nsage: M.representation_number_list(100)[99]==0\nFalse\n```\nBut of course if I represent 99 over the integers, then I have to represent it locally.  In this case the second output it correct, and the first one is incorrect.  I think there is some underlying problem with the way global quadratic forms are defined.  \n\n**Keywords:** bug\n\nIssue created by migration from https://trac.sagemath.org/ticket/20180\n\n",
    "closed_at": "2018-05-18T17:16:26Z",
    "created_at": "2016-03-08T21:00:18Z",
    "labels": [
        "component: quadratic forms",
        "critical",
        "bug",
        "wontfix"
    ],
    "title": "Bug: is_locally_represented_number_at_place returns incorrect value",
    "type": "issue",
    "url": "https://github.com/sagemath/sage/issues/20180",
    "user": "https://github.com/annahaensch"
}
```
Sorry this bug report is a bit ad hoc, but I suspect the thing that makes this buggy is part of a larger problem in the quadratic forms module.  I'm running Sage 6.9 and I get the following in error if I construct the quadratic form 

```
M=QuadraticForm(ZZ,3,[75,120,120,144,0,144])
```
then 

```
sage: M.is_locally_represented_number_at_place(99,3)
False
```
and 

```
sage: M.representation_number_list(100)[99]==0
False
```
But of course if I represent 99 over the integers, then I have to represent it locally.  In this case the second output it correct, and the first one is incorrect.  I think there is some underlying problem with the way global quadratic forms are defined.  

**Keywords:** bug

Issue created by migration from https://trac.sagemath.org/ticket/20180





---

archive/issue_events_182694.json:
```json
{
    "actor": "https://github.com/annahaensch",
    "created_at": "2017-10-22T21:30:31Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182694"
}
```



---

archive/issue_comments_292543.json:
```json
{
    "body": "<a id='comment:1'></a>\nThis is a duplicate of #23412 and can be closed",
    "created_at": "2017-10-22T21:30:31Z",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/20180#issuecomment-292543",
    "user": "https://github.com/annahaensch"
}
```

<a id='comment:1'></a>
This is a duplicate of #23412 and can be closed



---

archive/issue_events_182695.json:
```json
{
    "actor": "https://github.com/adeines",
    "created_at": "2017-10-23T20:46:34Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182695"
}
```



---

archive/issue_events_182696.json:
```json
{
    "actor": "https://github.com/adeines",
    "created_at": "2017-10-23T20:46:34Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "needs work",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182696"
}
```



---

archive/issue_comments_292544.json:
```json
{
    "body": "<a id='comment:3'></a>\nHey, I don't know what happened, but I didn't change this.",
    "created_at": "2017-10-23T22:02:09Z",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/20180#issuecomment-292544",
    "user": "https://github.com/adeines"
}
```

<a id='comment:3'></a>
Hey, I don't know what happened, but I didn't change this.



---

archive/issue_events_182697.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2018-03-10T12:49:31Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "needs work",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182697"
}
```



---

archive/issue_events_182698.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2018-03-10T12:49:31Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182698"
}
```



---

archive/issue_events_182699.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2018-03-10T12:49:31Z",
    "event": "demilestoned",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "milestone": "sage-7.1",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182699"
}
```



---

archive/issue_comments_292545.json:
```json
{
    "body": "<a id='comment:4'></a>\nduplicate of #23412, please close",
    "created_at": "2018-03-10T12:49:31Z",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/20180#issuecomment-292545",
    "user": "https://github.com/fchapoton"
}
```

<a id='comment:4'></a>
duplicate of #23412, please close



---

archive/issue_events_182700.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2018-03-10T12:49:47Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "needs review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182700"
}
```



---

archive/issue_events_182701.json:
```json
{
    "actor": "https://github.com/fchapoton",
    "created_at": "2018-03-10T12:49:47Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182701"
}
```



---

archive/issue_events_182702.json:
```json
{
    "actor": "https://github.com/videlec",
    "created_at": "2018-05-18T17:16:26Z",
    "event": "unlabeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "positive review",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182702"
}
```



---

archive/issue_events_182703.json:
```json
{
    "actor": "https://github.com/videlec",
    "created_at": "2018-05-18T17:16:26Z",
    "event": "closed",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182703"
}
```



---

archive/issue_comments_292546.json:
```json
{
    "body": "<a id='comment:6'></a>\nclosing positively reviewed duplicates",
    "created_at": "2018-05-18T17:16:26Z",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "type": "issue_comment",
    "url": "https://github.com/sagemath/sage/issues/20180#issuecomment-292546",
    "user": "https://github.com/videlec"
}
```

<a id='comment:6'></a>
closing positively reviewed duplicates



---

archive/issue_events_182704.json:
```json
{
    "actor": "https://github.com/videlec",
    "created_at": "2018-05-18T17:16:26Z",
    "event": "labeled",
    "issue": "https://github.com/sagemath/sage/issues/20180",
    "label": "wontfix",
    "type": "issue_event",
    "url": "https://github.com/sagemath/sage/issues/20180#event-182704"
}
```
